### PR TITLE
fix(DTFS2-6526): focus user to the table header button if it was clicked just now

### PR DESCRIPTION
**Pull request by avaitonis**
## Introduction :pencil2:
Activating any table header button will cause the table column to become filtered by ascending or descending. Keyboard-only users will be repositioned to the beginning of the main content, and this includes screen reader users. This could be problematic for the user as the user will need to re-navigate to the table content.

## Resolution :heavy_check_mark:
- Added hidden form input `formId` to detect what form was just activated. Was it sorting, search or pagination. If `formId` is `deals-table-sorting` then we know that sorting event just happened and we can `autofocus` sort column button.
- Added `sortButtonWasClicked` NJK variable to know if this is new sorting event happened.
- Ensured that autofocus doesn't happen for `default` sort and other navigations, like pagination and search.

## Miscellaneous :heavy_plus_sign:
Simplified how we get active `order` direction, `asc` or `desc`. This allows to have more form inputs controllers/deals/index.js line 61

### CODE REVIEW

File: trade-finance-manager-ui/server/controllers/deals/index.js [hunk 1]
```diff
@@ -48,8 +49,7 @@ const queryDeals = async (req, res) => {
   let activeSortByField = '';
   let searchString = '';
   const { userToken } = req.session;
-
-  delete req.body._csrf;
+  const sortButtonWasClicked = (req.body.formId && req.body.formId === 'deals-table-sorting') ? true : false;
```
Comment (abhi-markan) [inline trade-finance-manager-ui/server/controllers/deals/index.js:52]: Can be simplified to following statement

```javascript
 const sortButtonWasClicked = req.body?.formId === 'deals-table-sorting'
```

File: trade-finance-manager-ui/server/controllers/deals/index.js [hunk 2]
```diff
@@ -58,13 +58,12 @@ const queryDeals = async (req, res) => {
   const queryParams = { searchString };
 
   if (req.body.descending || req.body.ascending) {
-    const sortBy = Object.getOwnPropertyNames(req.body)[0];
-    const sortByField = req.body[sortBy];
-    activeSortByField = sortByField;
+    const order = req.body.ascending ? 'ascending' : 'descending';
```
Comment (abhi-markan) [inline trade-finance-manager-ui/server/controllers/deals/index.js:61]: Might be ideal to use optional chaining

```javascript
  const order = req.body?.ascending ? 'ascending' : 'descending';
```

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Few minor comments to address please.

**Review by vamshidharukef — DISMISSED** [review]
LGTM

File: trade-finance-manager-ui/server/controllers/deals/index.js (hunk 1 above)
Comment (avaitonis) [reply trade-finance-manager-ui/server/controllers/deals/index.js]: Thanks, simplified.

File: trade-finance-manager-ui/server/controllers/deals/index.js (hunk 2 above)
Comment (avaitonis) [reply trade-finance-manager-ui/server/controllers/deals/index.js]: Thanks, changed.

File: trade-finance-manager-ui/server/controllers/deals/index.js [hunk 3]
```diff
@@ -48,8 +49,7 @@ const queryDeals = async (req, res) => {
   let activeSortByField = '';
   let searchString = '';
   const { userToken } = req.session;
-
-  delete req.body._csrf;
+  const sortButtonWasClicked = (req.body?.formId === 'deals-table-sorting') ? true : false;
```
Comment (abhi-markan) [inline trade-finance-manager-ui/server/controllers/deals/index.js:52]: const sortButtonWasClicked = req.body?.formId

**Review by abhi-markan — CHANGES_REQUESTED** [review]
One minor amend please.

**Review by abhi-markan — APPROVED** [review]

File: trade-finance-manager-ui/server/controllers/deals/index.js (hunk 3 above)
Comment (avaitonis) [reply trade-finance-manager-ui/server/controllers/deals/index.js]: I simplified line 52.

Resolving this comment as PR was approved 1 hour ago.

**Pull request merged**